src/crewai/llm.py

```diff
@@ -371,6 +374,15 @@ def _handle_streaming_response(
         last_chunk = None
         chunk_count = 0
         usage_info = None
+        tool_calls = None
+        accumulated_tool_args: DefaultDict[int, SimpleNamespace] = defaultdict(
+            lambda: SimpleNamespace(
+                function=SimpleNamespace(
+                    name="",
+                    arguments="",
+                ),
+            ),
+        )
```

Comment:
Would it be overkill to use Pydantic here? Or a dataclass?

Reply:
good call!